File: js/romanNumerals.js
function toRoman(num){	
    let result = '';	
    // map of all possible combinations / divisors	
    const map = {	
      M: 1000,	
      CM: 900,	
      D: 500,	
      CD: 400,	
      C: 100,	
      XC: 90,	
      L: 50,	
      XL: 40,	
      X: 10,	
      IX: 9,	
      V: 5,	
      IV: 4,	
      I: 1,	
    };	
    // iterate through the map	
    for (key in map){	
      // add key for each time number is divisible by key value	
      result += key.repeat(Math.floor(num / map[key]));	
      // new number = remainder from previous calculation	
      num %= map[key];	
    }	
    return result;	
  }

  console.log(toRoman(2439)); // should return MMCDXXXIX

File: python/roman_numerals.py
def to_roman(num):
    result = ''
    I = 1
    V = 5
    X = 10
    L = 50
    C = 100
    D = 500
    M = 1000
    while num > 0:
        if num >= 1000:
            result = result + 'M'
            num = num - 1000
        elif num >= 500:
            result = result + 'D'
            num = num - 500
        elif num >= 100:
            result = result + 'C'
            num = num - 100
        elif num >= 50:
            result = result + 'L'
            num = num - 50
        elif num >= 10:
            result = result + 'X'
            num = num -10
        elif num >= 5:
            result = result + 'V'
            num = num - 5
        elif num >= 1:
            result = result + "I"
            num = num - 1
    return(result)

print(to_roman(2022))

# def to_roman(num):
#     result = ''
